docs/source/contributing/documentation.rst

```diff
@@ -0,0 +1,316 @@
+====================
+Adding Documentation
+====================
+
+When submitting a new class through a PR, or any changes in general,
+there should be documentation where possible. Here are the guidelines
+for writing it.
+
+Formatting and Running Tests
+----------------------------
+
+Please begin the description of the class/function in the same line as
+the 3 quotes:
+
+.. code:: py
+
+    def do_this():
+        """This is correct.
+        (...)
+        """
+
+    def dont_do_this():
+        """
+        This is incorrect.
+        (...)
+        """
+
+NumPy Format
+------------
+
+Use the numpy format for sections and formatting - see
+https://numpydoc.readthedocs.io/en/latest/format.html.
+
+This includes:
+
+1. The usage of ``Attributes`` to specify ALL ATTRIBUTES that a
+   class can have, their respective types and a brief (or long, if
+   needed) description. (See more on :ref:`_types`)
+
+Also, ``__init__`` parameters should be specified as ``Parameters`` **on
+the class docstring**, *rather than under* ``__init__``. Note that this
+can be omitted if the parameters and the attributes are the exact same
+(i.e., dataclass) - priority should be given to the ``Attributes``
+section, in this case, which must **always be present**, unless the
+class specifies no attributes at all. (See more on Parameters in number
+2 of this list.)
+
+Example:
+
+.. code:: py
+
+    class MyClass:
+        """My cool class. Long or short (whatever is more appropriate) description here.
+
+        Parameters
+        ----------
+        name : :class:`str`
+            The class's name.
+        id : :class:`int`
+            The class's id.
+        mobj : Optional[:class:`~.Mobject`], optional
+            The mobject linked to this instance. Defaults to `Mobject()` \
+    (is set to that if `None` is specified).
```

Comment:
Should we explicitly mention that instead of specifying types in the docstring, using type hints would be preferred?

Reply:
I agree that we should urge developers to enforce #52. I'll do a draft of this in a future commit to make the language more direct on our preference.